vue-client/src/components/mixins/sidesearch-mixin.vue

```diff
@@ -136,16 +127,49 @@ export default {
       this.fetch(n);
     },
     getItems(keyword) {
-      let params = this.getSearchParams(this.getSearchPhrase(keyword));
-      params = Object.assign(params, {
+      const searchPhrase = this.getSearchPhrase(keyword);
+      const urlSearchParams = new URLSearchParams({
+        q: searchPhrase,
         _limit: this.maxResults,
         _offset: this.currentPage * this.maxResults,
         _sort: this.sort,
+        ...(typeof this.typeArray !== 'undefined' && this.typeArray.length > 0 && { '@type': this.typeArray }),
+        ...this.currentSearchParam?.mappings,
+        ...this.currentSearchParam?.searchProps.reduce((acc, searchProp) => ({
+          ...acc,
+          [searchProp]: searchPhrase,
+        }), {}),
       });
-      if (typeof this.typeArray !== 'undefined' && this.typeArray.length > 0) {
-        params['@type'] = this.typeArray;
+
+      const field = VocabUtil.getTermObject(this.fieldKey, this.resources.vocab, this.resources.context);
+
+      /**
+       * If field is a kbv:predicate (e.g. role) then filter linkable items depending on field and parent type.
+       * */
+      if (field.subPropertyOf.find(subProp => subProp['@id'] === VocabUtil.getTermObject('predicate', this.resources.vocab, this.resources.context)['@id'])) {
```

Comment:
```suggestion
      // A VocabUtil.isSubPropertyOf(field.subPropertyOf, 'predicate', ...) would be preferable here.
      if (field.subPropertyOf.find(subProp => subProp['@id'] === VocabUtil.getTermObject('predicate', this.resources.vocab, this.resources.context)['@id'])) {
```
In the actual case here, this works. But the base property (here `predicate`) *could* be higher up in a `subPropertyOf` chain. I don't see a `VocabUtil.isSubPropertyOf` alas, so we could just note that this would be preferable in a comment. (We can factor out such method (from `getDomainList` around lines 442-450 in `vocab.js`) and use  in a separate PR.)